lib/base_service.js

```diff
@@ -157,9 +157,17 @@ BaseService.prototype._sanitizeConfig = function(conf, options) {
         // the number of workers has been supplied
         // on the command line, so honour that
         conf.num_workers = options.num_workers;
-    } else if (conf.num_workers === 'ncpu' || typeof conf.num_workers !== 'number') {
-        // use the number of CPUs
-        conf.num_workers = os.cpus().length;
+    }
+    if (typeof conf.num_workers !== 'number') {
+        if (/^ncpu\s*[\*\+\-\/]\s*\d+(?:\.\d+)?$/.test(conf.num_workers)) {
```

Comment:
1. Because it is conceivable that somebody wants an expression like `(ncpu + 1) / 2`, which the current code doesn't capture.
2. Not again, but instead of. Currently only expressions like `ncpu [+-*/] number` are allowed, which is just one type of expressions
3. Yup, but both amount to the same thing :)